From: Natanael Copa <ncopa@alpinelinux.org>
To: musl@lists.openwall.com
Subject: [musl] [PATCH] elf.h: add typedefs for Elf*_Relr

Add typedefs for Elf32_Relr and Elf64_Relr as a follow-up to commit
d32dadd60efb (ldso: support DT_RELR relative relocation format)

---
This fixes build of iproute2 with elfutils 0.190, which assumes that
Elf*_Relr are typedef'ed when SHT_RELR is defined.

ref: https://sourceware.org/git/?p=elfutils.git;a=commit;h=39f2c500542f69c2f1a13fd0ae4eaa5778d2ed8d
ref: https://sourceware.org/bugzilla/show_bug.cgi?id=31034

 include/elf.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/elf.h b/include/elf.h
index 23f2c4bc..aa186d9d 100644
--- a/include/elf.h
+++ b/include/elf.h
@@ -32,6 +32,9 @@ typedef uint16_t Elf64_Section;
 typedef Elf32_Half Elf32_Versym;
 typedef Elf64_Half Elf64_Versym;
 
+typedef Elf32_Word Elf32_Relr;
+typedef Elf64_Xword Elf64_Relr;
+
 #define EI_NIDENT (16)
 
 typedef struct {
-- 
2.42.1
